harness/determined/pytorch/trainer.py

```diff
@@ -0,0 +1,138 @@
+import contextlib
+import pathlib
+import warnings
+from enum import Enum
+
+import determined as det
+from determined import core, horovod, util
+from determined import pytorch
+from typing import Any, Callable, Dict, Iterator, List, Optional
+import torch
+import torch.nn as nn
+from determined.common import check
+from determined.horovod import hvd
+import torch.distributed as dist
+import logging
+
+from determined.profiler import ProfilerAgent
+from determined.pytorch._pytorch_trial import Batch, Epoch
+from determined.pytorch import PyTorchTrialContext, PyTorchTrialController
+
+
+class Trainer:
+    def __init__(self, trial: pytorch.PyTorchTrial, context: PyTorchTrialContext):
+        self.trial = trial
+        self.context = context
+        self.cluster_info = det.get_cluster_info()
+        self.core_context = self.context._core
+        self.distributed_backend = det._DistributedBackend()
+        self.det_profiler = ProfilerAgent._dummy_init()
+        self.local_training = self.cluster_info is None
+
+    def configure_profiler(self, sync_timings: bool, enabled: bool, begin_on_batch: int, end_after_batch: int):
```

Comment:
should this be attached to the Trainer?  Or part of the Train call?

Reply:
originally all of the profiler configs were part of the train() call, but I decided to separate them because i suspect the profiler might change down the road and we should keep the train() call relatively stable.